python/deltalake/writer.py

```diff
@@ -264,6 +265,24 @@ def write_deltalake(
         storage_options.update(storage_options or {})
 
         table.update_incremental()
+        num_indexed_cols = table._table.get_num_index_cols()
+        stats_cols = table._table.get_stats_columns()
+    else:
+        num_indexed_cols = DEFAULT_DATA_SKIPPING_NUM_INDEX_COLS
+        stats_cols = None
+        if configuration is not None:
+            if "delta.dataSkippingNumIndexedCols" in configuration:
+                str_indexed_cols = configuration["delta.dataSkippingNumIndexedCols"]
+                if str_indexed_cols is not None:
+                    num_indexed_cols = int(str_indexed_cols)
+            string_stats_cols = configuration.get("delta.dataSkippingStatsColumns")
+            if isinstance(string_stats_cols, str):
+                stats_cols = string_stats_cols.split(",")
+                if stats_cols and isinstance(stats_cols, list):
+                    if not all(isinstance(inner, str) for inner in stats_cols):
+                        raise ValueError(
+                            "dataSkippingStatsColumns needs to be a comma-separated list of column names in string format. Example: 'foo,bar,baz'"
+                        )
```

Comment:
is there a way for us to isolate reading the configuration in the rust side. as kernel progresses we will likely see a lot of this move there, and this way we have to keep multiple code locations in sync.

Reply:
I have added the exact same code in Rust. What I can do is move that into a pub fn, that I can call from python